src/Processors/QueryPlan/AggregatingStep.cpp

```diff
@@ -351,7 +353,8 @@ void AggregatingStep::transformPipeline(QueryPipelineBuilder & pipeline, const B
             return std::make_shared<AggregatingTransform>(header, transform_params, many_data, counter++, merge_threads, temporary_data_merge_threads);
         });
 
-        pipeline.resize(1);
+        /// No need to do this in case of aggregating in order, since AIO don't use two-level hash tables and thus returns only buckets with bucket_number = -1.
```

Comment:
```suggestion
        /// We add the explicit resize here, but not in case of aggregating in order, since AIO don't use two-level hash tables and thus returns only buckets with bucket_number = -1.
```